File: frameworks/dotnet/carter/Implementation/Routes/TemplateRoutes.cs
using Carter;
using Implementation.Components;
using Microsoft.AspNetCore.Http.HttpResults;

namespace Implementation.Routes;

/// <summary>
/// template: Carter has no view layer of its own, and what ASP.NET Core ships for server-side
/// HTML is Razor. RazorComponentResult renders a component statically from an endpoint, and
/// the Web SDK compiles the component into the assembly at build.
/// </summary>
public sealed class TemplateRoutes : ICarterModule
{
    public void AddRoutes(IEndpointRouteBuilder app)
    {
        app.MapGet("/template/small", (Payloads p) => Page(p.Small));

        app.MapGet("/template/medium", (Payloads p) => Page(p.Medium));
    }

    // rb:wiring template.*
    private static RazorComponentResult<ItemsPage> Page(Payload payload) =>
        new(new Dictionary<string, object?> { [nameof(ItemsPage.Body)] = payload });
}

File: frameworks/java/micronaut/Implementation/src/main/java/implementation/routes/TemplateRoutes.java
package implementation.routes;

import implementation.Payload;
import implementation.Payloads;
import io.micronaut.http.MediaType;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.views.ModelAndView;

/**
 * template: Micronaut Views with Thymeleaf. A handler returns a ModelAndView, and micronaut-views'
 * body writer has Thymeleaf render views/items-page.html with the payload's properties as the
 * model.
 */
@Controller
public class TemplateRoutes {

    private final Payloads p;

    TemplateRoutes(Payloads p) {
        this.p = p;
    }

    @Get(value = "/template/small", produces = MediaType.TEXT_HTML)
    public ModelAndView<Payload> small() {
        return page(p.small());
    }

    @Get(value = "/template/medium", produces = MediaType.TEXT_HTML)
    public ModelAndView<Payload> medium() {
        return page(p.medium());
    }

    // rb:wiring template.*
    /** The view Thymeleaf finds as views/items-page.html, and its model. */
    private static ModelAndView<Payload> page(Payload payload) {
        return new ModelAndView<>("items-page", payload);
    }
}
